src/enkf/gridio_fv3reg.f90

```diff
@@ -659,496 +661,1825 @@ subroutine writegriddata(nanal1,nanal2,vars3d,vars2d,n3d,n2d,levels,ndim,vargrid
+          enddo
+        endif !iope == 0 
+
+    endif
+
+    if (qnr_ind > 0) then
+       varstrname = 'rain_nc'
+       call fv3lamfile%get_idfn(varstrname,file_id,fv3filename)
+       call read_fv3_restart_data3d(varstrname,fv3filename,file_id,locworkvar3d)
+       do k=1,nlevs
+          call mpi_gatherv(locworkvar3d(1:nxloc,1:nyloc,k), recvcounts2d(iope+1), mpi_real4, &
+                      workvar3d(:,:,k), recvcounts2d, displs2d,&
+                      mpi_real4, 0,iocomms(mem_pe(nproc)) ,iret)
+       enddo
+       if(iope == 0 ) then
+         do k=1,nlevs
+            nn = nn_tile0
+            do j=1,ny_res
+            do i=1,nx_res
+               nn=nn+1
+               vargrid(nn,levels(qnr_ind-1)+k,nb,ne)=workvar3d(i,j,k)
+            enddo
+            enddo
+         enddo
+         do k = levels(qnr_ind-1)+1, levels(qnr_ind)
+             if (nproc .eq. 0)                                               &
+                write(6,*) 'READFVregional : qnr ',                           &
+                    & k, minval(vargrid(:,k,nb,ne)), maxval(vargrid(:,k,nb,ne))
+         enddo
+       endif !iope == 0 
+
+    endif
+   
+    ! set SST to zero for now
+    if (sst_ind > 0) then
+      if(iope == 0 ) then
+       vargrid(:,levels(n3d)+sst_ind,nb,ne) = zero
+      endif !iope == 0 
+    endif
+
+
+    !----------------------------------------------------------------------
+    ! Allocate memory for variables computed within routine
+ 
+    if (ps_ind > 0) then
+       if (iope ==0) then 
+         allocate(workprsi(nx_res,ny_res,nlevsp1))
+         allocate(pswork(nx_res,ny_res))
+       endif
+       call fv3lamfile%get_idfn('delp',file_id,fv3filename)
+       call read_fv3_restart_data3d('delp',fv3filename,file_id,locworkvar3d)  
+       do k=1,nlevs
+         call mpi_gatherv(locworkvar3d(1:nxloc,1:nyloc,k), recvcounts2d(iope+1), mpi_real4,&
+                   workvar3d(:,:,k), recvcounts2d, displs2d,&
+                   mpi_real4, 0,iocomms(mem_pe(nproc)) ,iret)
+       enddo
+       if(iope == 0 ) then
+         workprsi(:,:,nlevsp1)=eta1_ll(nlevsp1) !etal_ll is needed
+         do i=nlevs,1,-1
+           workprsi(:,:,i)=workvar3d(:,:,i)*0.01_r_kind+workprsi(:,:,i+1)
+         enddo
+    
+         pswork(:,:)=workprsi(:,:,1)
+
+
+
+         nn = nn_tile0
+         do j=1,ny_res
+            do i=1,nx_res
+               nn=nn+1
+               vargrid(nn,levels(n3d)+ps_ind, nb,ne) =pswork(i,j) 
+            enddo
+         enddo
+
+        if(nproc==0) then 
+                write(6,*) 'READFVregional : ps ',                           &
+                    & k, minval(vargrid(:,levels(n3d)+ps_ind,nb,ne)), maxval(vargrid(:,k,nb,ne))
+
+       endif !nproc==0  
+         
+
+         
+         do k=1,nlevs
+           do j=1,ny_res  
+            do i=1,nx_res
+              workvar3d(i,j,k)=(workprsi(i,j,k)+workprsi(i,j,k+1))*half
+            enddo
+           enddo
+         enddo
```

Comment:
Please align line 1692 with line 1688 and ensure at least two-space indentation for do loops.